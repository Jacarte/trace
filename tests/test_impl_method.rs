extern crate trace;

#[macro_use]
mod trace_test;

use trace::trace;

trace::init_depth_var!();

trace_test!(test_impl_method, {
    let foo = Foo;
    Foo::foo(2);
    foo.bar(7);
});

struct Foo;
impl Foo {
    fn foo(b: i32) -> i32 {
        b
    }

    #[trace]
    fn bar(&self, a: i32) -> i32 {
        a
    }
}
