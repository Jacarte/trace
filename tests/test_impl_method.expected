[+] Entering bar(a = 7)
[-] Exiting bar = 7
